### PR TITLE
Include timezone info in all dagster console log timestamps

### PR DESCRIPTION
**Pull request by gibsondan**
Summary:
Before I go in and fix the large number of tests this is likely to break, I wanted to see if this is something we even want to do. Adding timezone info to logs (both the dagit/daemon logs, but also run logs that are logged oto console)seems like a useful thing to me.

Concretely, before this change console output during a run might look something like:

`
2021-12-07 23:22:17 - dagster - DEBUG - asset_lineage_pipeline - 8b862d6a-4a75-424e-a52f-08455e7f5b37 - 6145 - daily_top_action - STEP_SUCCESS - Finished execution of step "daily_top_action" in 104ms.
`

Now, it will look like:
`
2021-12-07 23:22:17-0600 - dagster - DEBUG - asset_lineage_pipeline - 8b862d6a-4a75-424e-a52f-08455e7f5b37 - 6145 - daily_top_action - STEP_SUCCESS - Finished execution of step "daily_top_action" in 104ms.
`

### CODE REVIEW

**Comment by gibsondan** [comment]
Current dependencies on/for this PR:
* master
  * **PR #5897** 
    * **PR #5901**  (this PR)

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh) and will continue to be automatically updated while this PR remains open.

**Comment by prha** [comment]
yeah, I have no problems with this at all

**Review by alangenfeld — COMMENTED** [review]
are there string interpretations of time zones that are more legible? 
I guess this is the most accurate and consistent way to annotate, just feels a little hard to parse to me. 

If this is the standard best practice 100% ignore me.

**Comment by gibsondan** [comment]
i'm not sure what's standard per se but this is the one I find the clearest (as opposed to needing to look up something like Europe/Saratov on the map)

**Comment by sryza** [comment]
This seems like a reasonable space-to-clarity tradeoff to me.

File: python_modules/dagster/dagster/utils/log.py [hunk 1]
```diff
@@ -222,8 +223,12 @@ def default_format_string():
     return "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
 
 
+def default_date_format_string():
+    return "%Y-%m-%d %H:%M:%S%z"
```
Comment (alangenfeld) [inline python_modules/dagster/dagster/utils/log.py:227]: I think something like a space, parens or some delimiting for the `%z` would probably alleviate my minor visual parsing difficulty. 

Don't feel that strongly though so follow your heart

**Comment by gibsondan** [comment]
maybe we pick our favorite from one of the several example RFC specs here: https://pendulum.eustace.io/docs/#common-formats

**Review by alangenfeld — COMMENTED** [review]
interesting, seems like were closest to iso8601 so ill vote that

**Comment by sryza** [comment]
I find the `T` separator between date and time to be very difficult to parse visually.

**Review by alangenfeld — APPROVED** [review]
seems like this ` ` variant is rough consensus?
